### ISSUE TITLE
UI: Condense/rollup trades which include multiple orders by identical time stamp

### ISSUE THREAD

**Issue by ladidan** [rotki/rotki#3543]
## Abstract

The idea is to clean-up the trades UI in a way that trades which consist of multiple orders are condensed/rolled-up in one line.

## Motivation

Rotki users who are less interested in each single order which was executed but more keen on the bottom-line of a single trade could benefit from a much slimmer trades overview register.

## Specification

e.g. user A buys 1 ETH for x USD spot in one trade which is executed in 6 orders with separate IDs but all share the same time stamp (which is the the key differentiator)
The condensed trade consisting of multiple orders is summarized in one line and by clicking on it a drill-down of all single orders belonging to it shows up below.
I imagine a settings/option one-click checkmark for enable/disable this function.

**Comment by kelsos** [comment]
@LefterisJP we could make it an optional non-default behavior but is there an easy way to properly identify these trades?

**Comment by LefterisJP** [comment]
> @LefterisJP we could make it an optional non-default behavior but is there an easy way to properly identify these trades?

Depends on the exchange. We don't really do anything to differentiate but for some exchanges we could probably do more work (probably not too easy) to show them if the exchange can help us.

It may be possible in a hacky way using the "link" part as the common identifier is probably already there. Otherwise we would need to change the structure around a bit and this is too big a refactor to do at this point for a little benefit.

**Comment by LefterisJP** [comment]
> I want this feature to be available in Rotki so much!
> 
> Having a single human choice, an action, to buy some digital asset with some fiat is a unitary thing (in the mind of the human). It is only the exchange/order book mechanics in software that break the single human choice into many "trades" on an exchange.
> 
> It is probably not the _only_ way to do it. A tax accountant or whatever might need/care about the detail.
> 
> But a human, placing a market order that gets broken into eight trades on a centralized exchange, wants to see a single unitary trade: how much asset x was received in exchange for some amount of asset y, and what was the average price of that action.

An accountant already has all of the information they need. An order can be broken down into multiple trades that get executed over a period of weeks. Not having the individual trades is actually a bug as price can vary greatly in the in-between and that can affect cost basis calculation.

If an accountant wants to sum trades all they need is in the spreadsheet already. More data is good. Less data is bad.

**Comment by A2be** [comment]
Depends on the "user".  For the accountant, and tax accounting, you are correct.

But for the Rotki end user (me,, for example) who used some USD to buy some ETH a few days ago, I don't give a rip about all that detail once I've skimmed it.  Just want the dollars out, the ETH in, and the average price I actually achieved after I made that single unitary choice to execute a market trade.

**Comment by LefterisJP** [comment]
Yes this features make sense mostly as a "nice to have" for the UI. There I can see a point.

**Comment by kelsos** [comment]
Closing this one out.

Since this was filed, rotki moved from the old flat trades table to the history-events / swap model, which partially addresses the UX motivation here: a single trade is now rendered as one condensed swap row (spend -> receive, with fees) that you can expand to drill into the individual legs, instead of separate lines per leg.

What this issue specifically asked for is not implemented, though: rolling up multiple *distinct* orders/fills that share a timestamp into a single aggregated line. Swap events are grouped per trade (by group_identifier, which comes from the exchange's per-fill trade id), so each fill of a market order stays as its own swap row.

We're not planning to aggregate distinct fills into one line, for the reason raised earlier in the thread: collapsing separate fills loses per-fill price/time detail that matters for cost-basis accuracy. Closing as won't-do given the swap layout already covers the readability side of things.

**Comment by A2be** [comment]
Your solution makes sense.  Appreciate the clarity on not doing the meta-ask.

Still, the way Rotki shows events today is better than it was at the time @ladidan and @A2be wrote this up.  :)
